File: OneDrive/Documents/Data_Science/challenges/30-days-of-code/JavaScript/Day3_Prime_Number_Check.js
function isPrime(num) {
    if (num <= 1) {
        return false;
    }
    for (let i = 2; i <= Math.sqrt(num); i++) {
        if (num % i === 0) {
            return false;
        }
    }
    return true;
}

const num = prompt("Enter a number: ");
if (isPrime(num)) {
    console.log(`${num} is a prime number.`);
} else {
    console.log(`${num} is not a prime number.`);
}

File: OneDrive/Documents/Data_Science/challenges/30-days-of-code/GO/Day3_Prime_Number_Check.go
package main

import (
    "fmt"
    "math"
)

func isPrime(num int) bool {
    if num <= 1 {
        return false
    }
    sqrt := int(math.Sqrt(float64(num)))
    for i := 2; i <= sqrt; i++ {
        if num%i == 0 {
            return false
        }
    }
    return true
}

func main() {
    var num int
    fmt.Print("Enter a number: ")
    fmt.Scan(&num)

    if isPrime(num) {
        fmt.Printf("%d is a prime number.\n", num)
    } else {
        fmt.Printf("%d is not a prime number.\n", num)
    }
}
